pandas/tests/test_index.py

```diff
@@ -1304,6 +1304,12 @@ def test_symmetric_diff(self):
         self.assertTrue(tm.equalContents(result, expected))
         self.assertEqual(result.name, 'new_name')
 
+    def test_modulo(self):
+        # GH 9244
+        intidx = Index(range(10))
+        expected = Int64Index([0, 1, 0, 1, 0, 1, 0, 1, 0, 1], dtype='int64')
+        self.assert_index_equal(intidx % 2, expected)
+
```

Comment:
you are adding this test in a generic section, so it is not doing what you think. instead define in the ``TestNumeric`` section and use ``.create_index``. As I think this works for more index types.